File: app/src/main/java/net/north101/android/ghplayertracker/data/StatusSet.java
package net.north101.android.ghplayertracker.data;

import org.parceler.Parcel;
import org.parceler.ParcelConstructor;

@Parcel
public class StatusSet {
    public boolean disarm;
    public boolean stun;
    public boolean immobilize;
    public boolean poison;
    public boolean wound;
    public boolean muddle;
    public boolean invisible;
    public boolean strengthen;

    @ParcelConstructor
    public StatusSet(boolean disarm, boolean stun, boolean immobilize, boolean poison, boolean wound, boolean muddle, boolean invisible, boolean strengthen) {
        this.disarm = disarm;
        this.stun = stun;
        this.immobilize = immobilize;
        this.poison = poison;
        this.wound = wound;
        this.muddle = muddle;
        this.invisible = invisible;
        this.strengthen = strengthen;
    }

    public StatusSet() {}
}

File: app/src/main/java/net/north101/android/ghplayertracker/data/StatusSet.kt
package net.north101.android.ghplayertracker.data

import android.os.Parcelable
import kotlinx.android.parcel.Parcelize
import net.north101.android.ghplayertracker.RecyclerItemCompare

@Parcelize
class StatusSet(
        var disarm: Boolean = false,
        var stun: Boolean = false,
        var immobilize: Boolean = false,
        var poison: Boolean = false,
        var wound: Boolean = false,
        var muddle: Boolean = false,
        var invisible: Boolean = false,
        var strengthen: Boolean = false
) : Parcelable, RecyclerItemCompare {
    override val compareItemId: String
        get() = ""

}
